Formula/aws-sam-cli-nightly.rb

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+require_relative '../ConfigProvider/config_provider'
+
+class AwsSamCliNightly < Formula
+  include Language::Python::Virtualenv
+
+  config_provider = ConfigProvider.new('aws-sam-cli-nightly')
+
+  desc "AWS SAM CLI 🐿 is a tool for local development and testing of Serverless applications"
+  homepage "https://github.com/awslabs/aws-sam-cli/"
+  url config_provider.url()
+  sha256 config_provider.sha256
+  head "https://github.com/awslabs/aws-sam-cli.git", :branch => "nightly"
+
+  conflicts_with 'aws-sam-cli-rc', :because => "both install the 'sam' binary"
+  conflicts_with 'aws-sam-cli', :because => "both install the 'sam' binary"
```

Comment:
Why are we naming our nightly release the same as our stable? I don't think we should be naming our nightly CLI `sam`. Customers will not be able to install both, which I thought was a requirement of our nightly releases.

Reply:
Resolving the conflicts between two binaries will be complicated and the binary with different name will unstable, especially for Windows. We are not going to use them at the same time. Referring to different nightly release packages, they are using the same binary name. To distinct the nightly version from the stable version, we are having different nightly version number.